src/NzbDrone.Core/Http/ProxyHttpInterceptor.cs

```diff
@@ -0,0 +1,39 @@
+﻿using NzbDrone.Common.Http;
+using NzbDrone.Core.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace NzbDrone.Core.Http
+{
+    public class ProxyHttpInterceptor : IHttpRequestInterceptor
+    {
+        private readonly IConfigFileProvider _configService;
+
+        public ProxyHttpInterceptor(IConfigFileProvider configService)
+        {
+            this._configService = configService;
+        }
+
+        public HttpResponse PostResponse(HttpResponse response)
+        {
+            return response;
+        }
+
+        public HttpRequest PreRequest(HttpRequest request)
+        {
+            if(_configService.ProxyEnabled == UseProxy.Global ||
+                (_configService.ProxyEnabled == UseProxy.IndexersOnly && request.IsIndexerRequest))
+            {
+                request.Proxy = new HttpRequestProxySettings(_configService.ProxyType,
+                    _configService.ProxyHostname,
+                    _configService.ProxyPort,
+                    _configService.ProxyUsername == string.Empty ? null : _configService.ProxyUsername,
```

Comment:
there are times that you check against `string.empty` and times you check against `null` please use `String.IsNullOrWhiteSpace`